pl_examples/domain_templates/computer_vision_fine_tuning.py

```diff
@@ -303,59 +290,55 @@ def validation_step(self, batch, batch_idx):
         val_loss = self.loss(y_logits, y_true)
         num_correct = torch.eq(y_bin.view(-1), y_true.view(-1)).sum()
 
-        return {'val_loss': val_loss,
-                'num_correct': num_correct}
+        return {"val_loss": val_loss, "num_correct": num_correct}
 
     def validation_epoch_end(self, outputs):
         """Compute and log validation loss and accuracy at the epoch level."""
 
-        val_loss_mean = torch.stack([output['val_loss']
-                                     for output in outputs]).mean()
-        val_acc_mean = torch.stack([output['num_correct']
-                                    for output in outputs]).sum().float()
-        val_acc_mean /= (len(outputs) * self.batch_size)
-        return {'log': {'val_loss': val_loss_mean,
-                        'val_acc': val_acc_mean,
-                        'step': self.current_epoch}}
+        val_loss_mean = torch.stack([output["val_loss"] for output in outputs]).mean()
+        val_acc_mean = torch.stack([output["num_correct"] for output in outputs]).sum().float()
+        val_acc_mean /= len(outputs) * self.batch_size
+        self.log_dict({"val_loss": val_loss_mean, "val_acc": val_acc_mean, "step": self.current_epoch})
 
     def configure_optimizers(self):
-        optimizer = optim.Adam(filter(lambda p: p.requires_grad,
-                                      self.parameters()),
-                               lr=self.lr)
+        optimizer = optim.Adam(filter(lambda p: p.requires_grad, self.parameters()), lr=self.lr)
```

Comment:
if we just pass `self.parameters()` here, it won't update the ones with `requires_grad=False` right? so do we need to manually filter here?

Reply:
I am not sure about this particular one (neither do i know how to test it ....). I believe it is written that way to have the parameters added when the milestones are reached in `on_epoch_start`, by `_unfreeze_and_add_param_group`.

I will go ahead and remove it if you feel like it would still have the desired behavior.

Reply:
ok, let's leave it as it is. I didn't see that modified hook. Also, I think this might be better since Adam is adaptive and stores gradient info, so lesser the params, the lower will be the memory usage for grads.